enumerables.rb

```diff
@@ -0,0 +1,127 @@
+# rubocop: disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    n = 0
+    while n <= size - 1
+      yield(to_a[n])
+      n += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    n = 0
+    while n <= size - 1
+      yield(to_a[n], n)
+      n += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result = []
+    my_each do |n|
+      result.push(n) if yield(n)
+    end
+    result
+  end
+
+  def my_all?(var = nil)
+    result = true
+    my_each do |n|
+      if block_given?
+        result = false unless yield(n)
+      elsif var.nil?
+        result = false unless n
+      else
+        result = false unless var == n
+      end
+    end
+    result
+  end
+
+  def my_any?(var = nil)
+    result = false
+    my_each do |n|
+      if block_given?
+        return result = true if yield(n)
+      elsif var.nil?
+        return result = true unless n
+      else
+        return result = true unless var == n
+      end
+    end
+    result
+  end
+
+  def my_none?(var = nil)
+    result = true
+    my_each do |n|
+      if block_given?
+        return result = false if yield(n)
+      elsif var.nil?
+        return result = false unless n
+      else
+        return result = false unless var == n
+      end
+    end
+    result
+  end
+
+  def my_count(var = nil)
+    result = []
+    if block_given?
+      my_each do |n|
+        result.push(n) if yield(n)
+      end
+    elsif var.nil?
+      return length
+    else
+      my_each do |n|
+        result.push(n) if n == var
+      end
+    end
+    result.length
+  end
+
+  def my_map(var = nil)
+    result = []
+    return to_enum unless block_given? && var.nil?
+
+    if var
+      my_each do |n|
+        result.push(var.call(n))
+      end
+    else
+      my_each do |n|
+        result.push(yield(n))
+      end
+    end
+    result
+  end
+
+  def my_inject(var = nil)
+    return to_enum unless block_given?
+
+    if var
+      result = var
+    elsif var.nil?
+      result = self[0]
+    end
+    my_each do |n|
+      result = yield(result, n)
+    end
+    result
+  end
```

Comment:
* [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_inject combines all elements of enum by applying a binary operation, specified by a block
  ```ruby
    array = [4,7,8,89,9,6,5,12,4,2,5]
    operation = proc { |sum, n| sum + n  }
    p array.my_inject(&operation) == array.inject(&operation) # false
  ```
* [ ] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method
  ```ruby
    array = [1,7,8,89,9,6,5,12,4,2,5]
    p array.my_inject(:+) == array.inject(:+) # false
  ```

Reply:
Added symbol as a named method of inject.